enhancements/srvkp-2021-downstream-osp-on-pr.md

```diff
@@ -0,0 +1,153 @@
+# [SRVKP-2021](https://issues.redhat.com/browse/SRVKP-2021) — OSP "downstream" Operator per PR
+
+###### tags: `openshift-pipelines` `downstream`
+
+*End goal is to have this document in https://github.com/openshift-pipelines/enhancements*
+
+## Motivation
+
+The main motivation is to **provide a quick way to test changes in our product** before a change is merged upstream, and thus *without having to go through the whole release process*.
+
+- This should help QE to try things as early as possible
+- This should help developers know how their changes upstream will act in our product
+- This allow a quick feedback loop while testing or developing.
+
+## Design
+
+As described in [SRVKP-2021](https://issues.redhat.com/browse/SRVKP-2021), this can be done in several steps. We'll initially focus on inputs and outputs, independently of the medium used (slack, …). We'll use the simplest possible medium (a HTTP webhook, powered by [tektoncd/triggers](https://github.com/tektoncd/triggers) or knative), letting the more advanced use cases (slack bot, GitHub PRs, …) to a later document.
+
+In a nutshell, the idea is :
+
+- An HTTP service is exposed on a cluster (ospqa)
+- This services takes some inputs (what to build, where to push)
+- This services returns the pipelinerun name and the unique name to identify the build later
```

Comment:
1. shall we add a contraint to this proposal that we will use productized openshift-pipelines for implementation.
2. instead of returning the name of a pipelinerun shall we return the url to pipelinerun view (logs or pipeinerun viz) one OpenShift devCOnsole) :)

Reply:
> shall we add a contraint to this proposal that we will use productized openshift-pipelines for implementation.

What do you mean ? We shouldn't even care if it's OSP or upstream tektoncd/pipeline running for it, as long as it's something giving us the API 😁 

> instead of returning the name of a pipelinerun shall we return the url to pipelinerun view (logs or pipeinerun viz) one OpenShift devCOnsole) :)

Maybe both ? The main reason for getting the `pipelinerun` on *its own*, is to be able to automate interaction with it, like something that would wait for the `pipelinerun` to succeed before going further, …